scripts/restore.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+
+# Backup file directory path
+BACKUP_DIRECTORY_PATH="/palworld/backups"
+
+# Resotre path
+RESTORE_PATH="/palworld/Pal"
+
+# shellcheck disable=SC2317
+term_error_handler() {
+    echo "An error occurred during server shutdown."
+    exit 1
+}
+
+# shellcheck disable=SC2317
+restore_error_handler() {
+    printf "\033[0;31mAn error occurred during restore.\033[0m\n"
+    if [ -d "./tmp_save/Saved" ]; then
+        read -rp "I have a backup before recovery can proceed. Do you want to recovery it? (y/n): " RUN_ANSWER
+        if [[ $RUN_ANSWER == "y" ]] || [[ $RUN_ANSWER == "Y" ]]; then
+            rm -rf "$RESTORE_PATH/Saved"
+            mv "./tmp_save/Saved" "$RESTORE_PATH"
+            printf "\e[0;32mRecovery complete.\e[0m\n"
+        fi
+    fi
+
+    rm -rf "./tmp"
+    rm -rf "./tmp_save"
+
+    exit 1
+}
+
+if [ "${RCON_ENABLED}" != true ]; then
+    echo "RCON is not enabled. Please enable RCON to use this feature."
+    exit 1
+fi
+
+# Show up backup list
+echo "Backup List:"
+select BACKUP_FILE in "$BACKUP_DIRECTORY_PATH"/*; do
+    if [ -n "$BACKUP_FILE" ]; then
+        echo "Selected backup: $BACKUP_FILE"
+        break
+    else
+        echo "Invalid selection. Please try again."
+    fi
+done
+
+if [ -f "$BACKUP_FILE" ]; then
+    echo "Do you want to continue with the command?"
+    read -rp "When you run it, the server will be stopped and the recovery will proceed. (y/n): " RUN_ANSWER
+    if [[ $RUN_ANSWER == "y" ]] || [[ $RUN_ANSWER == "Y" ]]; then
+        printf "\e[0;32m*****STARTING PROCESS*****\e[0m\n"
+        # Shutdown server
+        trap 'term_error_handler' ERR
+
+        if [ "${RCON_ENABLED}" = true ]; then
+            printf "\e[0;32m*****SHUTDOWN SERVER*****\e[0m\n"
+            rcon-cli -c /home/steam/server/rcon.yaml save
+            rcon-cli -c /home/steam/server/rcon.yaml "shutdown 1"
+        else
+            echo "RCON is not enabled. Please enable RCON to use this feature. Unable to restore backup."
+            exit 1
+        fi
+        printf "\e[0;32mShutdown complete.\e[0m\n"
+
+        trap - ERR
+
+        trap 'restore_error_handler' ERR
+        
+        printf "\e[0;32m*****START RESTORE*****\e[0m\n"
+
+        # Recheck the backup file
+        if [ -f "$BACKUP_FILE" ]; then
+            # Copy the save file before restore
+            if [ -d "$RESTORE_PATH/Saved" ]; then
+                echo "Saves the current state before the restore proceeds."
+                rm -rf "./tmp_save"
+                mkdir -p "./tmp_save"
+                \cp -rf "$RESTORE_PATH/Saved" "./tmp_save/Saved"
```

Comment:
Instead of making a two fixed temporary directory I'd recommend using mktemp to make a parent for them in backups otherwise it's inside wherever you called the script from: `tmp_parent=$(mkdir -d -p /palworld/backup)`. Or do `tmp_parent="/palworld/backup/restore-"$(date +"%Y-%m-%d_%H-%M-%S")`.